app/src/main/java/com/jalasoft/routesapp/ui/adminPages/cities/view/CityAdminDetailFragment.kt

```diff
@@ -0,0 +1,175 @@
+            if (successResult) {
+                val message = if (isNew) RoutesAppApplication.resource?.getString(R.string.register_success).toString() else RoutesAppApplication.resource?.getString(R.string.edited_success).toString()
+                showProgress(false)
+                Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
+                findNavController().navigateUp()
+            } else {
+                showProgress(false)
+            }
+        }
+        viewModel.errorMessage.observe(this, errorObserver)
+        viewModel.successResult.observe(this, successResult)
+    }
+
+    private fun loadData(city: City?) {
+        val name = city?.name ?: ""
+        binding.etName.text = name.toEditable()
+        binding.etLat.text = city?.lat?.toEditable()
+        binding.etLng.text = city?.lat?.toEditable()
+    }
+
+    private fun buttonActions() {
+        binding.spCategory.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+                if (isNew) {
+                    val item = viewModel.cityCountry.value?.get(position)
+                    viewModel.countrySelected = item?.name ?: ""
+                    viewModel.countryID = item?.id ?: ""
+                } else {
+                    val item = editCityList[position]
+                    viewModel.countrySelected = item.name
+                    viewModel.countryID = item?.id ?: ""
+                }
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+                if (isNew) {
+                    viewModel.countrySelected = ""
+                } else {
+                    viewModel.countrySelected = city?.idCountry ?: ""
+                }
+            }
+        }
+
+        binding.btnEditSave.setOnClickListener {
+            if (isNew) {
+                showProgress(true)
+                val name = binding.etName.text.toString()
+                val lat = binding.etLat.text.toString()
+                val lng = binding.etLng.text.toString()
+                val selectedCountry = (binding.spCategory.selectedItem as Country).name
+                val countyId = viewModel.cityCountry?.value?.get(binding.spCategory.selectedItemPosition)?.id
+                viewModel.saveNewCity(name, lat, lng, selectedCountry, countyId ?: "0")
+            } else {
+                showProgress(true)
+                val name = binding.etName.text.toString()
+                val lat = binding.etLat.text.toString()
+                val lng = binding.etLng.text.toString()
+                val selectedCountry = (binding.spCategory.selectedItem as Country).name
+                val countyId = viewModel.cityCountry?.value?.get(binding.spCategory.selectedItemPosition)?.id
+                viewModel.updateCity(name, lat, lng, selectedCountry, countyId ?: "0", city!!)
```

Comment:
don't force to wrap the city, validate it firts

Reply:
Corrected :)